src/components/SearchModal/CurrencySearch.tsx

```diff
@@ -100,34 +97,57 @@ export function CurrencySearch({
     )
   }, [chainId, data?.portfolios])
 
-  const sortedTokens: Token[] = useMemo(
-    () =>
-      !balancesAreLoading
-        ? filteredTokens
-            .filter((token) => {
-              if (onlyShowCurrenciesWithBalance) {
-                return balances[token.address?.toLowerCase()]?.usdValue > 0
-              }
+  const sortedTokens: Token[] = useMemo(() => {
+    const portfolioTokens = data?.portfolios?.[0].tokenBalances
+      ?.map((tokenBalance) => {
+        if (!tokenBalance?.token?.chain || !tokenBalance.token?.address || !tokenBalance.token?.decimals) {
+          return undefined
+        }
+        return new Token(
+          supportedChainIdFromGQLChain(tokenBalance.token?.chain) ?? ChainId.MAINNET,
+          tokenBalance.token?.address,
+          tokenBalance.token?.decimals,
+          tokenBalance.token?.symbol,
+          tokenBalance.token?.name
+        )
+      })
+      .filter((token) => !!token) as Token[]
+
+    const filteredTokens = Object.values(defaultTokens)
+      .filter(getTokenFilter(debouncedQuery))
+      // Filter out tokens with balances so they aren't duplicated when we merge below.
+      .filter((token) => !(token.address?.toLowerCase() in balances))
+    const mergedTokens = [...(portfolioTokens ?? []), ...filteredTokens]
+
+    if (balancesAreLoading) {
+      return mergedTokens
+    }
```

Comment:
Moving this to the bottom is a behavior change. Should this be the first line of the `useMemo` callback?

Reply:
no because now we're using `mergedTokens` as the base list. look at this as if i added a bunch of logic before, and switched from a ternary to an early return , not as if i moved it.